### PR TITLE
Added details for the missing dotnet tool update command

### LINKED ISSUE #19368: Missing argument --version
`dotnet tool update` supports also the `--version` argument.

See also [this comment](https://github.com/dotnet/cli/pull/10205#issuecomment-506847148).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6a8171f4-8466-98ed-acb8-49563b864d16
* Version Independent ID: a1b7af2d-e2ba-bc4b-4d35-a8edbaa1249f
* Content: [dotnet tool update command - .NET Core CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-update)
* Content Source: [docs/core/tools/dotnet-tool-update.md](https://github.com/dotnet/docs/blob/master/docs/core/tools/dotnet-tool-update.md)
* Product: **dotnet-core**
* Technology: **dotnet-cli**
* GitHub Login: @tdykstra
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Added details for the missing `dotnet tool update` command, including:

- `--disable-parallel`
- `--ignore-failed-sources`
- `--interactive`
- `--local`
- `--no-cache`
- `--tool-manifest <PATH>`
- `--version <VERSION>`

Also, added examples demonstrating semantic versioning and link.

Fixes #19368

### Preview

✔️ [dotnet tool update](https://review.docs.microsoft.com/en-us/dotnet/core/tools/dotnet-tool-update?branch=pr-en-us-19369)

### CODE REVIEW

**Review by tdykstra — DISMISSED** [review]
Thanks for jumping in to take care of this.  I made some suggestions to make some things consistent with the CLI docs and clarify some of the text.

EDIT: the synopsis suggestions were meant to alphabetize the relevant options for each scenario, on second look they may have missed a line in each case.

File: docs/core/tools/dotnet-tool-update.md [hunk 1]
```diff
@@ -93,8 +129,21 @@ The `dotnet tool update` command provides a way for you to update .NET Core tool
...
+
+  Updates the [dotnetsay](https://www.nuget.org/packages/dotnetsay/) global tool to the latest version.
+
+- **`dotnet tool update -g dotnetsay --version 2.0.*`**
+
+  Updates the [dotnetsay](https://www.nuget.org/packages/dotnetsay/) global tool to the latest patch version, with a major version of `2`, and a minor version of `0`.
+
+- **`dotnet tool update -g dotnetsay --version (2.0.*,2.1.4)`**
+
+  Updates the [dotnetsay](https://www.nuget.org/packages/dotnetsay/) global tool to the lowest tolerable version within the specified range `(> 2.0.0 && < 2.1.4)`, version `2.1.0` would be installed.
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply docs/core/tools/dotnet-tool-update.md:141]: That's a good callout, it took me a while too - to dig into that.

File: docs/core/tools/dotnet-tool-update.md [hunk 2]
```diff
@@ -14,17 +14,24 @@ ms.date: 02/14/2020
...
-dotnet tool update <PACKAGE_NAME> --tool-path <PATH>
+dotnet tool update <PACKAGE_ID> --tool-path <PATH>
     [--configfile <FILE>] [--framework <FRAMEWORK>]
-    [-v|--verbosity <LEVEL>] [--add-source <SOURCE>]
+    [--add-source <SOURCE>] [--disable-parallel]
+    [--ignore-failed-sources] [--interactive] [--no-cache]
+    [-v|--verbosity <LEVEL>] [--version <VERSION>]
 
-dotnet tool update <PACKAGE_NAME>
+dotnet tool update <PACKAGE_ID>
```
Comment (tdykstra) [inline docs/core/tools/dotnet-tool-update.md:29]: I missed this earlier -- Now that we have a `--local` option we should put that on this line, corresponding to `--global` and `--tool-path` for the preceding two scenarios.

**Review by BillWagner — APPROVED** [review]
All of @tdykstra 's comments have been addressed. Approving while Tom is out on vacation.

**Pull request merged**